main.cpp

```diff
@@ -24,44 +37,62 @@ int main() {
 		if (!validcommand(parancs)) {
 			cout << "'" << parancs << "' is not recognized as an internal or external command, operable program or batch file." << endl;
 		}
-		if (parancs == "mkdir") {
-			string dirname;
-			cin >> dirname;
-			d.mkdir(dirname, CurrentFolder);
-		}
-		if (parancs == "cd") {
-			string dirname;
-			cin >> dirname;
-			if (dirname == "..") {
-				if (CurrentFolder != "root") {
-					int cut = CurrentFolder.find_last_of("/");
-					CurrentFolder = CurrentFolder.substr(0, cut);
-				}
-				else {
-					cout << "You're already in the root directory." << endl;
-				}
+		else {
+			if (parancs == "ls") {
+				d.ls(CurrentFolder);
 			}
 			else {
-				CurrentFolder = d.cd(dirname, CurrentFolder);
-			}
-		}
-
-		if (parancs == "ls") {
-			d.ls(CurrentFolder);
-		}
-		if (parancs == "rm") {
-			string dirname;
-			cin >> dirname;
-			if (dirname == "-rf") {
+				string dirname;
 				cin >> dirname;
-				d.rmForce(dirname, CurrentFolder);
+				vector<string> path;
+				if (dirname.find_first_of("/") < 4294967290) {
+					path = split(dirname);
+					path.pop_back();
+					dirname = dirname.substr(dirname.find_last_of("/") + 1, dirname.size());
```

Comment:
this is the same string that was deleted in the previous line from the end of `path`, isn't it?

Reply:
Így van. Igazából lehetett volna a vektor utolsó eleme is, de így is ugynaz az eredmény.